File: Disarium Number (Special Numbers Series #3)/Disarium Number (Special Numbers Series #3).dart
import "dart:math";

String disariumNumber(int n) {
  String sn = "${n}";
  int s = 0;
  
  for (int i = 0; i < sn.length; i++) {
    s = s + pow(int.parse(sn[i]),i+1).floor();
  }
  
  return s == n ? "Disarium !!" : "Not !!";
}

File: Disarium Number (Special Numbers Series #3)/Disarium Number (Special Numbers Series #3).swift
func disariumNumber(_ number: Int) -> String {
    let digits = String(number).compactMap { Int(String($0)) }
    let sum = zip(digits.indices, digits)
        .map { j, i in Int(pow(Double(i), Double(j + 1))) }
        .reduce(0, +)
    return sum == number ? "Disarium !!" : "Not !!"
}
